### ISSUE TITLE
Selections bug

### ISSUE THREAD

**Issue by ysard** [labsquare/cutevariant#78]
Selections doesn't select only visible variants:

Ex: 
- Take a db file for test.vep.vcf
- Let's imagine that a variant is referenced on chr and pos (chr,pos,ref,alt behaves the same anyway)
- Add a variant with same chr, pos as the 11th in the db:
`INSERT into variants (chr, pos, ref, alt) values (11, 920000, 'G', 'Y')`
- Imagine the following filtered request for this variant made by the gui:
`SELECT variants.id,chr,pos,ref,alt,COUNT(*) as 'children' FROM variants where pos = 920000 and alt = "Y" GROUP BY chr,pos`
=> We search 1 position and a specific alternative base, so we expect to get only 1 variant.

| id | chr | pos | ref | alt | children |
| -- | -- | -- | -- | -- | -- |
| 12 | 11 |	920000 | G | Y | 1 |

- Let's imagine a user wants to create a selection on this query with the actual code:
```
SELECT DISTINCT variants.id FROM variants
  INNER JOIN (
    SELECT variants.id,chr,pos,ref,alt,COUNT(*) as 'children' FROM variants where pos = 920000 and alt = "Y" GROUP BY chr,pos
  ) AS query
  ON variants.chr = query.chr
  AND variants.pos = query.pos
```
This returns the following variants ids:

| id |
| --- |
| 11 |
| 12 |

The code forgets all filters and I suppose that this is not we expect ?
We should rely only on variants ids.

Another problem:
 - Same as previous until the filter query:
`SELECT variants.id,chr,pos,ref,alt,COUNT(*) as 'children' FROM variants where pos = 920000 GROUP BY chr,pos`

| id | chr | pos | ref | alt | children |
| -- | -- | -- | -- | -- | -- |
| 12 | 11 |	920000 | G | Y | 2 |

Definition of a variant is still modified here: chr,pos instead of chr,pos,ref,alt.
The id column is a non-sense in this case because it is an arbitrary data selected outside the group by.
Here, the creation of the selection will be ok:

| id |
| --- |
| 11 |
| 12 |

Bilan:
- Sort variants on chr,pos instead of chr,pos,ref,alt is a real mess.
- We should never try to group variants on chr,pos in the gui.
- Selections should always take care of filters, so they should only rely on variants ids returned by the query which is used to create them.

**Comment by dridk** [comment]
Et si on enlève le GROUP BY ? Je veux dire par la, qu'on utilise le GROUP BY uniquement pour l'affichage.

**Comment by ysard** [comment]
Fixed now, see #74 for remaining ideas.
Group by chr,pos is currently fixed in my branch. Will see what's up with the next viewquerywidget.
